WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019 Western Digital Corporation or its affiliates.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//
+// Based on a public domain version of printf
+// Web link: http://www.rte.se/sites/default/files/Blog/Modesty/ee_printf.c 
+//
+
+
+#include <stdarg.h>
+#include "mem_map.h"
+#include "printf.h"
+
+
+/*---------------------------------------------------*/
+/* Define											 */ 
+/*---------------------------------------------------*/
+#define UART_RX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x0)))
+#define UART_TX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x4)))
+#define UART_STAT		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x8)))
+#define UART_CTRL 		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0xC)))
+
+#define UART_TX_BUSY	(1<<3)
+#define UART_RX_AVAIL	(1<<0)
+
+
+/*---------------------------------------------------*/
+/* strlen											 */
+/*---------------------------------------------------*/
+int strlen(const char *s)
+{
+    const char *it = s;
+
+	while(*it)
+	{
+		++it;
+	}
+    return it - s;
+}
+
+
+/*---------------------------------------------------*/
+/* uart_putchar: Write character to UART			 */
+/*---------------------------------------------------*/
+int uart_putchar(char ch)   
+{   
+    if (ch == '\n')
+        uart_putchar('\r');
+
+	//check status
+    while (UART_STAT & UART_TX_BUSY);
+	//write char
+    UART_TX_DATA = ch;
+
+    return 0;
+}
+
+/*----------------------------------------------------*/
+/* Use the following parameter passing structure to   */
+/* make printf re-entrant.                         */
+/*----------------------------------------------------*/
+typedef struct params_s 
```

Comment:
structure not following the coding convention

Reply:
yes, not in this phase